tklock.c

```diff
@@ -294,6 +304,81 @@ static void disable_autorelock_policy(void)
 	return;
 }
 
+/**
+ * Enable/disable X11 input events (currently only for touchscreen)
+ *
+ * @param enable TRUE enable events, FALSE disable events
+ * @return TRUE on success, FALSE on failure
+ */
+static gboolean generic_x11_event_control(const gboolean enable)
+{
+	gboolean ret = FALSE;
+	Display* dpy = NULL;
+	XDeviceInfo *devinfo = NULL;
+	int ndev = 0;
+	unsigned char data = (unsigned char)enable;
+
+	dpy = XOpenDisplay(NULL);
+	if (dpy == NULL) {
+		dpy = XOpenDisplay(":0.0");
+	}
+
+	if (dpy == NULL) {
+		mce_log(LL_INFO, "%s: unable to open display", __func__);
+		return ret;
+	}
+
+	if (x11_atom_touchscreen == None)
+		x11_atom_touchscreen = XInternAtom(dpy, XI_TOUCHSCREEN, True);
+
+	if (x11_atom_device_enabled == None)
+		x11_atom_device_enabled = XInternAtom(dpy, "Device Enabled", False);
+
+	if ((x11_atom_touchscreen == None) || (x11_atom_device_enabled == None)) {
+		mce_log(LL_WARN, "%s: unable to obtain X11 Atoms", __func__);
+		goto done;
+	}
+
+	devinfo = XListInputDevices(dpy, &ndev);
+
+	for (int i = 0; i < ndev; i++) {
+		XDeviceInfo info = devinfo[i];
+
+		if (info.use != IsXExtensionPointer)
```

Comment:
Shouldn't it be
`if (info.use != IsXPointer || info.use != IsXExtensionPointer)`?

Reply:
I took this logic from here: https://github.com/maemo-leste/hildon-desktop/blob/cursor_management/src/main.c#L415

Reply:
@spinal84 - the same question here - what makes you think IsXPointer is needed as well?

Reply:
According to XListInputDevices manual page:
```
The use field specifies how the device is currently being used.
If the value is IsXKeyboard, the device is a master keyboard.
If the value is IsXPointer, the device is a master pointer. If
the value is IsXExtensionPointer, the device is a slave
pointer.
```

Reply:
Do we want to track only slave pointer devices?

Reply:
I think we only want to work on slaves, since the goal is have to X11 not read input events *and* close(2) the backing fd, which it does, if we close the 'slave' pointer device. The naming is confusing for sure, but we also don't /want/ to disable master pointers.

```
The slave devices are the physical hardware devices which controls the
master devices. For example, you can see the Logitech Optical USB Mouse
listed as a slave pointer in the above list of my devices. The mouse is
one of two input pointer devices which is considered a slave to the
Virtual core pointer.
```